src/components/button/Button.tsx

```diff
@@ -41,23 +41,28 @@ export const Button: React.FC<ButtonProps> = ({
     prefix,
     `${prefix}--${variant}`,
     iconButton && `${prefix}--icon`,
-    iconRight && `${prefix}--icon-right`,
-    iconLeft && `${prefix}--icon-left`,
-    `${loading ? 'loading' : ''}`,
+    loading && 'loading',
     `${prefix}--${size}`
   );
-  if(variant==='destructive') {
+
+  if(variant === 'destructive') {
     console.warn('The button variant: \'destructive\' will be deprecated.\n Please use: \'primary-destructive\' instead')
   }
+
   return (
     <button
+      role="button"
```

Comment:
Just curious. Why do you need to add the role 'button' on a button tag? (See https://stackoverflow.com/questions/26934719/do-i-need-role-button-on-a-button)